Java/tests/src/test/java/com/microsoft/recognizers/text/tests/datetime/DateTimeParserTest.java

```diff
@@ -25,118 +27,121 @@
-                        Assert.assertEquals(getMessage(currentCase, "futureResolution"), expectedValue.getFutureResolution(), actualValue.getFutureResolution());
-                        Assert.assertEquals(getMessage(currentCase, "pastResolution"), expectedValue.getPastResolution(), actualValue.getPastResolution());
-                    }
-                });
-    }
-
-    private static IDateTimeParser getParser(TestCase currentCase) {
-        try {
-            String culture = getCultureCode(currentCase.language);
-            String name = currentCase.modelName;
-            switch (culture) {
-                case Culture.English:
-                    return getEnglishParser(name);
-                default:
-                    throw new AssumptionViolatedException("Parser Type/Name not supported.");
-            }
-        } catch (IllegalArgumentException ex) {
-            throw new AssumptionViolatedException(ex.getMessage(), ex);
-        }
-    }
-
-    private static IDateTimeParser getEnglishParser(String name) {
-        switch (name) {
-            case "DurationParser":
-                return new BaseDurationParser(new EnglishDurationParserConfiguration(new EnglishCommonDateTimeParserConfiguration(DateTimeOptions.None)));
-            case "DateParser":
-                return new BaseDateParser(new EnglishDateParserConfiguration(new EnglishCommonDateTimeParserConfiguration(DateTimeOptions.None)));
-            default:
-                throw new AssumptionViolatedException("Parser Type/Name not supported.");
-        }
-    }
-
-    private IDateTimeExtractor getExtractor(TestCase currentCase) {
-        String extractorName = currentCase.modelName.replace("Parser", "Extractor");
-        return DateTimeExtractorTest.getExtractor(currentCase.language, extractorName);
-    }
-
-    public static <T extends DateTimeResolutionResult> T parseDateTimeResolutionResult(Class<T> dateTimeResolutionResultClass, Object result) {
-        // Deserializer
-        ObjectMapper mapper = new ObjectMapper();
-        mapper.configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true);
-        mapper.addMixIn(DateTimeResolutionResult.class, DateTimeResolutionResultMixIn.class);
-        mapper.addMixIn(TimeZoneResolutionResult.class, TimeZoneResolutionResultMixIn.class);
-
-        try {
-            String json = mapper.writeValueAsString(result);
-            return mapper.readValue(json, dateTimeResolutionResultClass);
-
-        } catch (JsonProcessingException e) {
-            e.printStackTrace();
-            return null;
-
-        } catch (IOException e) {
-            e.printStackTrace();
-            return null;
-        }
-    }
+	private static final String recognizerType = "DateTime";
+
+	@Parameterized.Parameters(name = "{0}")
+	public static Collection<TestCase> testCases() {
+			return AbstractTest.enumerateTestCases(recognizerType, "Parser");
```

Comment:
Nit: Indentation seems weird throughout the file. Let's standardize on 4 whitespaces as in most of the code.